### ISSUE TITLE
Error: Transaction Rollback on forløbselement

### ISSUE THREAD

**Issue by Marc-Petersen** [scandihealth/lpr3-docs#306]
We're hit with a Error Detail: soap:Server>Transaction rolled back on many of our submissions.

There are no double observations (as far as I can see) like the similar issue posted earlier.

FE: 43D405B6-27DD-11E9-B969-005056B7A4B4.

What other information do you need?

**Comment by Marc-Petersen** [comment]
I spot-checked for ECPR (I'm not sure how to check for APPEND or REPLACE), but versionNumber is 2 (or higher), so your hypothesis sounds right.

I dont have a user nor an address for your SFTP :(
Could I put an anonymized CDA here as an attachment?

**Comment by TueCN** [comment]
Ok, we will continue working with that hypothesis. I don't think we need an anonymized version at this time then. 
We will try to reproduce the error in our internal environment (we have an idea about what might be wrong) and work on a fix. Hopefully this issue can be resolved quickly.

FYI: 
If the CDA `ClinicalDocument` element has a `relatedDocument` child element, then it is either an append or replace. Here is an example (typeCode="APND" means append):
```xml
<relatedDocument typeCode="APND">
    <parentDocument classCode="DOCCLIN" moodCode="EVN">
        <id root="9827ad00-ef98-4439-bca5-9503a5b817c8" extension="a3999026-e0b2-4804-846b-01db62d73d91"/>
        <setId root="9827ad00-ef98-4439-bca5-9503a5b817c8"/>
        <versionNumber value="2"/>
    </parentDocument>
</relatedDocument>
```

**Comment by jonigkeit** [comment]
Hi @Marc-Petersen

Can you post the value of the setId root attribute. It is a truly random identifier and would help us greatly.

It should look like this
```xml
 <hl7:setId root="3AAEB8CF-9BB3-4175-B586-A71BB36D1EDC"/>
```

**Comment by Marc-Petersen** [comment]
Alright. New CDA with the same error.
FE ID: BBF3803D-29E9-509D-AE93-4C2403379C35
Set ID: f3c582f4-2799-11e9-8b0a-005056b7a4b4

**Comment by TueCN** [comment]
@Marc-Petersen could you please attach an anonymized version of one of the failed requests anyway, however could you preserve the **root** values? (They are not sensitive information)

**Comment by jonigkeit** [comment]
The problem occurs when submitting a document that has a record target with multiple alternative identifiers that have overlapping root values, e.g.
```xml
    <hl7:recordTarget>
        <hl7:templateId root="2.16.840.1.113883.10.12.101"/>
        <hl7:templateId root="1.2.208.176.7.1.10.101"/>
        <hl7:patientRole classCode="PAT">
            <hl7:templateId root="1.2.208.176.7.1.10.107"/>
            <hl7:id root="81a94801-0b85-44b3-a7e5-3d1586da25a0" extension="e6a1a686-3548-4a8a-8622-781bc6efbb3f"/>
            <hl7:id root="81a94801-0b85-44b3-a7e5-3d1586da25a0" extension="b1e085e1-59ef-4d77-b9c4-1fb6e293b246"/>
            <hl7:addr use="H">
                <hl7:streetAddressLine nullFlavor="NI"/>
                <hl7:postalCode nullFlavor="NI"/>
                <hl7:city nullFlavor="NI"/>
                <hl7:country>5100</hl7:country>
            </hl7:addr>
            <hl7:patient classCode="PSN">
                <hl7:name>
                    <hl7:family>John</hl7:family>
                    <hl7:given>Doe</hl7:given>
                </hl7:name>
                <hl7:administrativeGenderCode codeSystem="2.16.840.1.113883.5.1" code="M"/>
            </hl7:patient>
        </hl7:patientRole>
    </hl7:recordTarget>
```

While the interface allows this it is not in agreement with the *Indberetningsvejledningen section 3.3.1 Patient og bopæl*

We will raise this issue with SDS.

**Comment by Marc-Petersen** [comment]
Any progress on this issue with SDS?

**Comment by KirstenLHSDS** [comment]
We have been requesting information from you (via Mads W) on when this happens, to be able to understand the usecase behind the registrations, creating this issue. Please help SDS on information about the registration, that causes this?

**Comment by Marc-Petersen** [comment]
Every example I have seen with multiple IDs, are when a patient has a ECPR and a (number of) historical ECPR(s) within our EPJ. All those patient IDs are included in the CDA.

**Comment by KirstenLHSDS** [comment]
Thanks, will it be different ECPRs (and in that case, why?)?

**Comment by Marc-Petersen** [comment]
Yes. Different ECPRs. Those are sent in the cases where patient changes ECPR (for reasons unknown to me) - thus logging a new current ECPR and adding the previous ECPR the pool of historical ECPRs within the patient record. All are sent to LPR.

**Comment by KirstenLHSDS** [comment]
Thank you for the input. SDS patient registry will have to consider, how this is supposed to be handled. It may be until end of next week, until an answer is ready.

**Comment by KirstenLHSDS** [comment]
SDS answers that, Indberetningsvejledningen is valid and thus, that only one patientID is expected in the CDA.
The answer - in danish is:
Der skal indberettes ét og kun et unikt patientID pr. indberetning (uanset om denne indeholder ét eller flere FE’ere), også selvom CDA’en giver mulighed for flere ID’er.

**Comment by DaveSS** [comment]
I would like to make sure I am clear on this resolution. Are you saying that there is just 1 patient ID in an LPR3 document, or 1 patient ID of a given type.

There is a specific template (1.2.208.176.7.1.10.133) for the case where a patient might have a both CPR identifier and an Alternative Identifier. Is this template no longer valid in an LPR3 document?
